### PR TITLE
Proposal: Iteration

### PR DESCRIPTION
**Pull request by Arthurdw**

### CODE REVIEW

File: concepts/iteration.coral [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Contains example iterations.
+*/
+
+// For loop
+for (let i = 0; condition; i++) {
+    ...
+}
+
+// Foreach loop
+foreach item in items {
```
Comment (Sculas) [inline concepts/iteration.coral]: All of the other iteration loops have parentheses around them. I think for consistency we should do that here too:
```suggestion
foreach (item in items) {
```

**Review by Sculas — CHANGES_REQUESTED** [review]

**Comment by Sculas** [comment]
Also, do we want to support `do-while` loops? They may be useful in some cases, and I don't see a reason to not support them. What do you think?

**Comment by Arthurdw** [comment]
> Also, do we want to support \`do-while\` loops? They may be useful in some cases, and I don't see a reason to not support them. What do you think?

We could, but I have never really needed to use it. So thought to not bloat the language. If you'd beg to differ, then I'll add it :)

**Comment by Sculas** [comment]
It's not needed now and may be added in the future.

**Review by Sculas — APPROVED** [review]

**Pull request merged**